conans/model/recipe_ref.py

```diff
@@ -135,6 +136,16 @@ def validate_ref(self):
         if self.channel and validation_pattern.match(self.channel) is None:
             raise ConanException(f"Invalid package channel '{self.channel}'")
 
+        # Warn if they use .+- in the name/user/channel, as it can be problematic for generators
+        pattern = re.compile(r'[.+-]')
+        if pattern.search(self.name):
+            ConanOutput().warning(f"Name containing special chars is discouraged '{self.name}'")
+        if self.user and pattern.search(self.user):
+            ConanOutput().warning(f"User containing special chars is discouraged '{self.user}'")
+        if self.channel and pattern.search(self.channel):
+            ConanOutput().warning(f"Channel containing special chars is discouraged "
+                                  f"'{self.channel}'")
```

Comment:
Could it be a bit noisy? I mean, how many times are those warnings appearing if the name contains any of those symbols?

Reply:
This appears only when a package is being exported, that is, ``conan export`` or ``conan create``. But not in every other usage, requirement, or anywhere else

Reply:
That's great, then.